### PR TITLE
add proxy question

### PR DESCRIPTION
**Pull request by knull-cn**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-dm) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)

add proxy question

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v3.0 changes)
- [x] v2.0 (TiDB DM 2.0 versions)
- [x] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/faq.md [hunk 1]
```diff
@@ -332,3 +332,14 @@ query-status test
 DM-master 会在启动时将 etcd 信息记录在当前目录。如果重启后当前目录发生变化，会导致 DM 缺失 etcd 信息，从而启动失败。
 
 推荐使用 TiUP 运维 DM 避免这一问题。在需要使用二进制部署的场合，需要在 DM-master 配置文件中使用绝对路径配置 data-dir 项，或者注意运行命令的当前目录。
+
+## dmctl 连接 DM-master 失败
```
Comment (lance6716) [inline zh/faq.md:336]:
```suggestion
## dmctl 连接 DM-master 失败

```

File: zh/faq.md [hunk 2]
```diff
@@ -332,3 +332,14 @@ query-status test
...
+在执行 `dmctl` 相关命令，发现连接 `dm-master` 失败（即使指定了 `--master-addr`），报错内容可能是“RawCause: context deadline exceeded, Workaround: please check your network connection.”。但是，检查网络却发现没有问题（比如，telnet x.x.x.x xx 正常）  
+
+这种情况可以检查下环境变量 `https_proxy` (注意，这里是 **https** )。如果配置了该环境变量， `dmctl` 会自动去连接 `https_proxy` ，而 `https_proxy` 主机没有相应的 `proxy` 转发服务，导致连接失败。
```
Comment (lance6716) [inline zh/faq.md:339]:
```suggestion
这种情况可以检查下环境变量 `https_proxy` (注意，这里是 **https** )。如果配置了该环境变量，`dmctl` 会自动去连接 `https_proxy` ，而 `https_proxy` 主机没有相应的 `proxy` 转发服务，导致连接失败。  
```

File: zh/faq.md [hunk 3]
```diff
@@ -332,3 +332,14 @@ query-status test
 DM-master 会在启动时将 etcd 信息记录在当前目录。如果重启后当前目录发生变化，会导致 DM 缺失 etcd 信息，从而启动失败。
 
 推荐使用 TiUP 运维 DM 避免这一问题。在需要使用二进制部署的场合，需要在 DM-master 配置文件中使用绝对路径配置 data-dir 项，或者注意运行命令的当前目录。
+
+## dmctl 连接 DM-master 失败
+在执行 `dmctl` 相关命令，发现连接 `dm-master` 失败（即使指定了 `--master-addr`），报错内容可能是“RawCause: context deadline exceeded, Workaround: please check your network connection.”。但是，检查网络却发现没有问题（比如，telnet x.x.x.x xx 正常）  
+
+这种情况可以检查下环境变量 `https_proxy` (注意，这里是 **https** )。如果配置了该环境变量， `dmctl` 会自动去连接 `https_proxy` ，而 `https_proxy` 主机没有相应的 `proxy` 转发服务，导致连接失败。  
+
+解决方案，可以去确认下 `https_proxy` 是否必须要配置，如果不是必须的，那么可以取消掉该设置即可。如果环境必须的，那么可以在原命令前加个环境变量设置：`https_proxy="" ./dmctl --master-addr "x.x.x.x:8261" `
```
Comment (lance6716) [inline zh/faq.md:341]:
```suggestion
解决方案，可以去确认下 `https_proxy` 是否必须要配置，如果不是必须的，那么可以取消掉该设置即可。如果环境必须的，那么可以在原命令前加个环境变量设置：`https_proxy="" ./dmctl --master-addr "x.x.x.x:8261"`  
```

File: zh/faq.md [hunk 4]
```diff
@@ -332,3 +332,14 @@ query-status test
...
+## dmctl 连接 DM-master 失败
+在执行 `dmctl` 相关命令，发现连接 `dm-master` 失败（即使指定了 `--master-addr`），报错内容可能是“RawCause: context deadline exceeded, Workaround: please check your network connection.”。但是，检查网络却发现没有问题（比如，telnet x.x.x.x xx 正常）  
+
+这种情况可以检查下环境变量 `https_proxy` (注意，这里是 **https** )。如果配置了该环境变量， `dmctl` 会自动去连接 `https_proxy` ，而 `https_proxy` 主机没有相应的 `proxy` 转发服务，导致连接失败。  
+
+解决方案，可以去确认下 `https_proxy` 是否必须要配置，如果不是必须的，那么可以取消掉该设置即可。如果环境必须的，那么可以在原命令前加个环境变量设置：`https_proxy="" ./dmctl --master-addr "x.x.x.x:8261" `  
+
+> **补充说明：**   
+> 
+> 关于 `proxy` 的环境变量有 `http_proxy` , `https_proxy` , `no_proxy` 等。如果这里连接有问题，可以考虑查看下这个几个参数看看有没有问题。
```
Comment (lance6716) [inline zh/faq.md:345]:
```suggestion
> 关于 `proxy` 的环境变量有 `http_proxy`, `https_proxy`, `no_proxy` 等。如果这里连接有问题，可以考虑查看下这个几个参数看看有没有问题。  
```

**Comment by Liuxiaozhen12** [comment]
Does this PR apply to v2.0 or not? PTAL, thanks! @knull-cn @lance6716

File: zh/faq.md [hunk 5]
```diff
@@ -332,3 +332,14 @@ query-status test
 DM-master 会在启动时将 etcd 信息记录在当前目录。如果重启后当前目录发生变化，会导致 DM 缺失 etcd 信息，从而启动失败。
 
 推荐使用 TiUP 运维 DM 避免这一问题。在需要使用二进制部署的场合，需要在 DM-master 配置文件中使用绝对路径配置 data-dir 项，或者注意运行命令的当前目录。
+
+## dmctl 连接 DM-master 失败
+在执行 `dmctl` 相关命令，发现连接 `dm-master` 失败（即使指定了 `--master-addr`），报错内容可能是“RawCause: context deadline exceeded, Workaround: please check your network connection.”。但是，检查网络却发现没有问题（比如，telnet x.x.x.x xx 正常）
```
Comment (Liuxiaozhen12) [inline zh/faq.md:337]:
```suggestion
在执行 `dmctl` 相关命令，发现连接 `dm-master` 失败（即使指定了 `--master-addr`），报错内容可能是 “RawCause: context deadline exceeded, Workaround: please check your network connection”。但是，检查网络却发现没有问题（比如，telnet x.x.x.x xx 正常）。
```

File: zh/faq.md (hunk 3 above)
Comment (Liuxiaozhen12) [inline zh/faq.md:341]:
```suggestion
解决方案：确认 `https_proxy` 是否必须要配置，如果不是必须的，取消该设置即可。如果环境必须，那么在原命令前加环境变量设置 `https_proxy="" ./dmctl --master-addr "x.x.x.x:8261" ` 即可。
```

File: zh/faq.md (hunk 4 above)
Comment (Liuxiaozhen12) [inline zh/faq.md:345]:
```suggestion
> 关于 `proxy` 的环境变量有 `http_proxy` , `https_proxy` , `no_proxy` 等。如果这里连接有问题，可以考虑查看这个几个参数设置是否有问题。  
```

File: zh/faq.md (hunk 5 above)
Comment (CharLotteiu) [inline zh/faq.md:337]:
```suggestion
在使用 dmctl 执行相关命令时，发现连接 `dm-master` 失败（即使已在命令中指定 `--master-addr` ），报错内容类似 `RawCause: context deadline exceeded, Workaround: please check your network connection.`，但使用 `telnet <master-addr>` 之类的命令检查网络却没有发现异常。  
```

File: zh/faq.md (hunk 2 above)
Comment (CharLotteiu) [inline zh/faq.md:339]:
```suggestion
这种情况可以检查下环境变量 `https_proxy`（注意，这里是 **https** ）。如果配置了该环境变量，dmctl 会自动去连接 `https_proxy`  指定的主机，而如果该主机没有相应的 `proxy` 转发服务，则会导致连接失败。  
```

File: zh/faq.md (hunk 4 above)
Comment (CharLotteiu) [inline zh/faq.md:345]:
```suggestion
> **注意：**   
> 
> 关于 `proxy` 的环境变量有 `http_proxy`，`https_proxy`，`no_proxy` 等。如果这里连接有问题，可以考虑查看下这个几个参数看看是否有问题。  
```
“有没有问题”具体是指什么问题？前面 L339 才强调了要看的是 `https_proxy`，所以是想表达如果修改了 `https_proxy` 后还是不能成功的话再检查一下其他两个环境变量吗？

File: zh/faq.md (hunk 1 above)
Comment (CharLotteiu) [inline zh/faq.md:336]:
```suggestion
## 使用 dmctl 执行命令时无法连接 DM-master

```

File: zh/faq.md (hunk 5 above)
Comment (lance6716) [reply zh/faq.md:337]: dmctl 也要反引号吗

File: zh/faq.md (hunk 2 above)
Comment (lance6716) [reply zh/faq.md:339]: 可以表述为 “dmctl 会自动去连接 `https_proxy` 指定的主机及端口”

File: zh/faq.md (hunk 5 above)
Comment (CharLotteiu) [reply zh/faq.md:337]: 不需要反引号，目前还是先听我的😄

Comment (lance6716) [reply zh/faq.md:338]: `dm-master` 应该也要去掉

Comment (CharLotteiu) [reply zh/faq.md:338]: Updated

File: zh/faq.md (hunk 2 above)
Comment (CharLotteiu) [reply zh/faq.md:340]: Updated

File: zh/faq.md (hunk 4 above)
Comment (lance6716) [reply zh/faq.md]: 这个由于原来的行被改动了，变成 outdated 了。可能需要重新提一下？

File: zh/faq.md [hunk 6]
```diff
@@ -332,3 +332,15 @@ query-status test
...
+
+在使用 dmctl 执行相关命令时，发现连接 DM-master 失败（即使已在命令中指定 `--master-addr` 的参数值），报错内容类似 `RawCause: context deadline exceeded, Workaround: please check your network connection.`，但使用 `telnet <master-addr>` 之类的命令检查网络却没有发现异常。  
+
+这种情况可以检查下环境变量 `https_proxy`（注意，这里是 **https** ）。如果配置了该环境变量，dmctl 会自动去连接 `https_proxy`  指定的主机及端口，而如果该主机没有相应的 `proxy` 转发服务，则会导致连接失败。  
+
+解决方案：确认 `https_proxy` 是否必须要配置，如果不是必须的，取消该设置即可。如果环境必须，那么在原命令前加环境变量设置 `https_proxy="" ./dmctl --master-addr "x.x.x.x:8261" ` 即可。
+
+> **补充说明：**   
+> 
+> 关于 `proxy` 的环境变量有 `http_proxy`, `https_proxy`, `no_proxy` 等。如果这里连接有问题，可以考虑查看这个几个参数设置是否有问题。
```
Comment (CharLotteiu) [inline zh/faq.md:346]:
```suggestion
> **注意：**   
> 
> 关于 `proxy` 的环境变量有 `http_proxy`，`https_proxy`，`no_proxy` 等。如果这里连接有问题，可以考虑查看下这个几个参数看看是否有问题。  
```
“有没有问题”具体是指什么问题呢？前面 L340 有强调要看的是 `https_proxy`，所以是想表达如果修改了 `https_proxy` 后还是不能成功的话再检查一下其他两个环境变量吗？

Comment (lance6716) [reply zh/faq.md:346]: 对的，背景是我们当时复现发现影响变量是 `https_proxy`，但怀疑用户是 `http_proxy` 影响的

File: zh/faq.md [hunk 7]
```diff
@@ -332,3 +332,15 @@ query-status test
...
 
 推荐使用 TiUP 运维 DM 避免这一问题。在需要使用二进制部署的场合，需要在 DM-master 配置文件中使用绝对路径配置 data-dir 项，或者注意运行命令的当前目录。
+
+## 使用 dmctl 执行命令时无法连接 DM-master
+
+在使用 dmctl 执行相关命令时，发现连接 DM-master 失败（即使已在命令中指定 `--master-addr` 的参数值），报错内容类似 `RawCause: context deadline exceeded, Workaround: please check your network connection.`，但使用 `telnet <master-addr>` 之类的命令检查网络却没有发现异常。  
+
+这种情况可以检查下环境变量 `https_proxy`（注意，这里是 **https** ）。如果配置了该环境变量，dmctl 会自动去连接 `https_proxy`  指定的主机及端口，而如果该主机没有相应的 `proxy` 转发服务，则会导致连接失败。  
+
+解决方案：确认 `https_proxy` 是否必须要配置，如果不是必须的，取消该设置即可。如果环境必须，那么在原命令前加环境变量设置 `https_proxy="" ./dmctl --master-addr "x.x.x.x:8261" ` 即可。
```
Comment (lance6716) [inline zh/faq.md:342]:
```suggestion
解决方案：确认 `https_proxy` 是否必须要配置，如果不是必须的，取消该设置即可。如果环境必须，那么在原命令前加环境变量设置 `https_proxy="" ./dmctl --master-addr "x.x.x.x:8261"` 即可。
```

File: zh/faq.md (hunk 6 above)
Comment (CharLotteiu) [reply zh/faq.md:346]: Suggestion updated. PTAL.

**Review by lance6716 — APPROVED** [review]

**Review by CharLotteiu — APPROVED** [review]

**Pull request merged**